SYCL/ESIMD/api/functional/functions/functions_select_2d.hpp

```diff
@@ -0,0 +1,286 @@
+          typename DimT>
+class run_test {
+  static constexpr int NumElems = DimT::value;
+  static constexpr int SizeY = SizeYT::value;
+  static constexpr int StrideY = StrideYT::value;
+  static constexpr int SizeX = SizeXT::value;
+  static constexpr int StrideX = StrideXT::value;
+  static constexpr int OffsetX = OffsetXT::value;
+  static constexpr int OffsetY = OffsetYT::value;
+  static constexpr int Height = NumElems / SimdSplitCoeffT::value;
+  static constexpr int Width = NumElems / Height;
+
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    bool passed = true;
+
+    static_assert(NumElems == Height * Width,
+                  "Invalid SimdSplitCoeff, height and width multiplacation "
+                  "should be equal to simd size.");
+    static_assert(Width > SizeX, "Number selected elements can't be greater "
+                                 "than number elements in one line");
+    static_assert(
+        Height > SizeY,
+        "Number selected elements can't be greater than lines number");
+    assert(Height > OffsetY &&
+           "Offset of height should be less than simd_view height size");
+    assert(Width > OffsetX &&
+           "Offset of with should be less than simd_view with size");
+
+    constexpr size_t value_for_increase_ref_data_for_fill = 50;
+    static_assert(std::numeric_limits<char>::max() >
+                      value_for_increase_ref_data_for_fill + NumElems,
+                  "Value that used for increase ref data for fill plus simd "
+                  "size should be less than char max value.");
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> result(NumElems, allocator);
+    shared_vector<DataT> initial_ref_data(NumElems, allocator);
+    shared_vector<DataT> ref_data_for_fill(SizeY * SizeX, allocator);
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      initial_ref_data[i] = i + 1;
+    }
+    // We should have different values in the first reference data and in the
+    // second reference data.
+    for (size_t i = 0; i < NumElems; ++i) {
+      ref_data_for_fill[i] =
+          initial_ref_data[i] + value_for_increase_ref_data_for_fill;
+    }
+
+    queue.submit([&](sycl::handler &cgh) {
+      DataT *init_ref_ptr = initial_ref_data.data();
+      DataT *ref_data_for_fill_ptr = ref_data_for_fill.data();
+      DataT *const out_ptr = result.data();
+      cgh.single_task<
+          Kernel<DataT, NumElems, TestCaseT, SizeYT, StrideYT, SizeXT, StrideXT,
+                 SimdSplitCoeffT, OffsetXT, OffsetYT>>([=]() SYCL_ESIMD_KERNEL {
+        TestCaseT::template call_esimd_function<DataT, NumElems, SizeY, StrideY,
+                                                SizeX, StrideX, Height, Width>(
+            init_ref_ptr, ref_data_for_fill_ptr, out_ptr, OffsetX, OffsetY);
+      });
+    });
+    queue.wait_and_throw();
+
+    std::vector<size_t> selected_indexsess;
```

Comment:
```suggestion
    std::vector<size_t> selected_indexes;
```
Isn't it the same misprint as we discussed in #939 and in https://github.com/intel/llvm-test-suite/pull/875#discussion_r825963443 ?

Reply:
Naming and other cases has been fixed in this commit: https://github.com/intel/llvm-test-suite/pull/908/commits/ce54fc2c173d7fdf6944b2e91334eba4dbf7f90c